src/lektorium/repo/local/repo.py

```diff
@@ -0,0 +1,218 @@
+import atexit
+import collections.abc
+import contextlib
+import datetime
+import functools
+import inifile
+import pathlib
+import shutil
+import tempfile
+
+import bidict
+import yaml
+from cached_property import cached_property
+
+from ..interface import (
+    Repo as BaseRepo,
+    DuplicateEditSession,
+    InvalidSessionState,
+    SessionNotFound,
+)
+from .config import Config
+
+
+class Site(collections.abc.Mapping):
+    ATTR_MAPPING = bidict.bidict({
+        'name': 'site_name',
+        'email': 'custodian_email',
+        'owner': 'custodian',
+    })
+
+    def __init__(self, site_id, production_url, **props):
+        self.data = dict(props)
+        restricted_keys = ('sessions', 'staging_url')
```

Comment:
Move `restricted_keys` to class constant
This will also allow to add length of `restricted_keys` rather than `2` in `__len__` method. And using `yield from restricted_keys` in `__iter__` looks better

Reply:
1. This constants highly related to the code below in init
2. Code in iter and len use different sets of constants